crates/storage/src/lean/db.rs

```diff
@@ -0,0 +1,61 @@
+use std::{fs, io, path::PathBuf, sync::Arc};
+
+use anyhow::Result;
+use redb::{Builder, Database};
+use tracing::info;
+
+use crate::errors::StoreError;
+
+pub const REDB_FILE: &str = "lean.redb";
+
+/// The size of the cache for the database
+///
+/// 1 GiB
+pub const REDB_CACHE_SIZE: usize = 1_024 * 1_024 * 1_024;
+
+#[derive(Clone, Debug)]
+pub struct ReamLeanDB {
+    pub db: Arc<Database>,
+    pub data_dir: PathBuf,
+}
+
+impl ReamLeanDB {
+    pub fn new(data_dir: PathBuf) -> Result<Self, StoreError> {
+        let db = Builder::new()
+            .set_cache_size(REDB_CACHE_SIZE)
+            .create(data_dir.join(REDB_FILE))?;
```

Comment:
We should only have this once

We should have 

```ReamDB``` which as a constructor to build ReamDB, then it will have 2 functions

BeaconDB, LeanDB these will hold network specific functions. These will only be constructable from `ReamDB`